src/main/java/onlinebookstore/model/User.java

```diff
@@ -2,22 +2,33 @@
 
 import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.JoinTable;
+import jakarta.persistence.ManyToMany;
 import jakarta.persistence.Table;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
 import lombok.Data;
 import lombok.RequiredArgsConstructor;
 import org.hibernate.annotations.SQLDelete;
 import org.hibernate.annotations.Where;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 @Data
 @Entity

```

Comment:
Using Lombok's `@Data` annotation on `@Entity` classes in Java, particularly in JPA (Java Persistence API) environments like Hibernate, is generally not recommended due to potential issues with equals, hashCode, and toString methods.
When you use `@Data`, Lombok generates these methods automatically, which can lead to problems in JPA environments. For instance, including all fields in equals and hashCode methods might lead to unexpected behavior when entities are managed by JPA. Additionally, the toString method might inadvertently trigger lazy loading of entity associations, causing performance issues or even unexpected behavior.
In JPA, it's often recommended to manually implement equals, hashCode, and toString methods in entity classes, customizing them based on business requirements and ensuring they don't interfere with the persistence mechanism. Therefore, while Lombok can save boilerplate code in many scenarios, it's best to avoid using `@Data` on JPA entity classes and instead selectively use Lombok annotations like `@Getter`, `@Setter`, `@EqualsAndHashCode`, and `@ToString` with caution and understanding of their implications in JPA environments.